airflow/providers/google/suite/hooks/calendar.py

```diff
@@ -0,0 +1,213 @@
+# under the License.
+#
+"""This module contains a Google Sheets API hook"""
+
+from datetime import datetime
+from typing import Any, Dict, Optional, Sequence, Union
+
+from googleapiclient.discovery import build
+
+from airflow.exceptions import AirflowException
+from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
+
+
+class GCalendarHook(GoogleBaseHook):
+    """
+    Interact with Google Calendar via Google Cloud connection
+    Reading and writing cells in Google Sheet:
+    https://developers.google.com/calendar/api/v3/reference
+
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param api_version: API Version
+    :type api_version: str
+    :param delegate_to: The account to impersonate using domain-wide delegation of authority,
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account.
+    :type impersonation_chain: Union[str, Sequence[str]]
+    """
+
+    def __init__(
+        self,
+        gcp_conn_id: str = 'google_cloud_default',
+        api_version: str = 'v3',
+        delegate_to: Optional[str] = None,
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+    ) -> None:
+        super().__init__(
+            gcp_conn_id=gcp_conn_id,
+            delegate_to=delegate_to,
+            impersonation_chain=impersonation_chain,
+        )
+        self.gcp_conn_id = gcp_conn_id
+        self.api_version = api_version
+        self.delegate_to = delegate_to
+        self._conn = None
+
+    def get_conn(self) -> Any:
+        """
+        Retrieves connection to Google Calendar.
+
+        :return: Google Calendar services object.
+        :rtype: Any
+        """
+        if not self._conn:
+            http_authorized = self._authorize()
+            self._conn = build('calendar', self.api_version, http=http_authorized, cache_discovery=False)
+
+        return self._conn
+
+    def get_events(
+        self,
+        calendar_id: str = 'primary',
+        i_cal_uid: Optional[str] = None,
+        max_attendees: Optional[int] = None,
+        max_results: Optional[int] = None,
+        order_by: Optional[str] = None,
+        page_token: Optional[str] = None,
```

Comment:
I'm not sure if `page_token` should be an argument

Reply:
Removed. I agree .. we have a loop in the code to loop through all pages.